Review Comments

egekorkan opened the pull request:
Testing the method described here http://astrofrog.github.io/blog/2013/04/10/how-to-conduct-a-full-code-review-on-github/ , pointed by @zoltankis

ereshetova committed:
Rewriting S&P section without normative statements

Commit:
pubdate nov 29

Commit:
use HTML5.2 instead of whatwg living standard

Commit:
use HTML5.2 instead of whatwg living standard

Commit:
Changes at Appendix A

Commit:
Changes at Appendix A

Commit:
add how to subscribe the group list

Commit:
update diff.html

Commit:
add data-editor-id

danielpeintner committed:
fix TS issue "Cannot find name 'USVString'" by replacing USVString with string

danielpeintner committed:
align with TD Security changes

* SecurityScheme: rename proxyURI to proxy
* Rename APIKeySecurityScheme to ApikeySecurityScheme
* BearerSecurityScheme: add attribute authorization
* CertSecurityScheme: missing
* Rename PskScheme to PSKSecurityScheme and add attribute identity
* PublicSecurityScheme: missing
* Rename PopSecurityScheme to PoPSecurityScheme
* OAuth2SecurityScheme: add attributes token and refresh

danielpeintner committed:
further TD spec alignments

change writable to readOnly and writeOnly
add MultiLanguage term for titles and descriptions
add safe and idempotent to Action

danielpeintner committed:
fix: add missing TypeScript terms (scopes,response) and fix terms (security) in Form Interface

add required ExpectedResponse interface

danielpeintner committed:
update definitions: add forms to top-level, fix security definitions, add base, lastModified, ..

Commit:
Add appendix for WebIDL index

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #159 from zolkis/master

Add appendix for WebIDL index

danielpeintner committed:
Merge branch 'master' of https://github.com/w3c/wot-scripting-api

danielpeintner committed:
fix: add scopes, uriVariables, and security terms to InteractionFragment (TypeScript definitions)

danielpeintner committed:
feat: add (experimental) support for passing options to property.read()/write() and action.invoke() (e.g., for uriVariables)

danielpeintner committed:
feat: add options to handlers also

danielpeintner committed:
fix: uriVariables is not of type DataSchema. Instead it is of map of DataSchema

danielpeintner committed:
refactor: get ready for node-wot release 0.6.0

danielpeintner committed:
refactor: update to node-wot 0.7.0-SNAPSHOT.1

Commit:
Merge pull request #155 from ereshetova/master

Rewriting S&P section without normative statements

Commit:
Reorganize the Security section

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #160 from zolkis/master

Reorganize the Security section

Commit:
Revise scripting use cases and terminology.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #161 from zolkis/master

Revise scripting use cases and terminology.

danielpeintner committed:
feat: start adapting new API

reading properties for consumed thing for now

danielpeintner committed:
feat: move new API calls to thingInstance

danielpeintner committed:
fix: string array issue

danielpeintner committed:
feat: add writing properties for new API

hasbel committed:
Fix #163

Change to the typescript definitions to make "items" in ArraySchema and "properties" in ObjectSchema non mandatory (as described by the TD spec).

danielpeintner committed:
fix: add invokeAction for next API

danielpeintner committed:
fix: TypeScript definitions links

Commit:
Update editors list, header, abstract and use cases

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Update the WOT object, fetching, parsing, instantiating a TD, consume(), produce(), discover(), examples.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Update ConsumeThing, ExposedThing, remove unnecessary sections, start writing some of the algorithms.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Update ExposedThing methods and algorithms, update examples, update TD parsing and instantiating steps, the Changes and Acknowledgements sections.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Add missing ConsumedThing method specifications.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #164 from hasbel/patch-1

Fix #163

Commit:
Editorial sweep: clarify sections, correct white spacing etc

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Refactor discovery without events

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Refine discovery, add more example. Improve text in first chapters.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Simplify discovery

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Clarify name for TD object and string. Remove TD change event and use case. Note on uriVariables.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #166: remove uriVariables from this API version

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Move discovery in its own section, separated from the core API

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Editorial fixes: move discovery use cases, add links, correct Thingweb

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #165 from zolkis/master

Rewrite the spec according to past few months discussions for the new WD

Commit:
Fix #169: Property observation

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
feat: add observeProperty/unobserveProperty as well as subscribeEvent/unsubscribeEvent

Commit:
Merge pull request #170 from zolkis/master

WiP: Editorial sweep

danielpeintner committed:
fix: align with latest CR thing-description changes

* rename "name" to "title"
* add "titles" to thing level
* change "Versioning" to "VersionInfo"
* change "lastModified" to "modified"

danielpeintner committed:
fix: remaining change "name" to "title"

danielpeintner committed:
Merge branch 'master' of https://github.com/w3c/wot-scripting-api

Commit:
Fix ReSpec error, add data-cite to Error

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #171: change name to title

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #173: separate client and server interfaces

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Remove ThingInstance and string ThingDescription

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Make constructible ConsumedThing, ExposedThing and ThingDiscovery primary API entry points

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Add UA conformance classes. Add WoT object and keep constructors. Editorial fixes

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #172: update security references

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Align conformance class names with Architecture spec

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Editorial sweep

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #174 from zolkis/master

Separate client and server API

danielpeintner committed:
chore: update typescript definitions version for next node-wot release

danielpeintner committed:
fix: align with TD "Versioning" --> "VersionInfo"

danielpeintner committed:
fix: align with TD ("name" --> "title")

Commit:
Correct Web IDL for new errors. Add ed note for iterables.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #179 from zolkis/master

Correct Web IDL for new errors. Add ed note for iterables.

danielpeintner committed:
feat: align with WoT interface

* rename WoTFactory to WoT
* adapt discover
* change consume using Promise
* change produce using Promise
* remove register and unregister from WoT interface

danielpeintner committed:
increase snapshot version

danielpeintner committed:
feat: more alignments with current scripting API
* remove fetch
* produce accepting TD string only
* remove ThingModel

danielpeintner committed:
feat: align with new scripting API

* ThingDescription object vs. string
* simplify/remove all other definitions

danielpeintner committed:
feat: define ThingDescription as any key/value map to avoid TD warnings/errors

Commit:
Fix #183: add id to observe and subscribe

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #184: change .td to function .getThingDescription() and add the ||td|| internal slot

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #182: re-add InteractionOptions that contain uriVariables, define PropertyValueMap

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #185 from zolkis/master

Handle implementation feedback from node-wot

Commit:
Fix #180: Clarify ExposedThing Property initialization

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
add .pr-preview.json

Commit:
Merge pull request #186 from zolkis/master

Fix #180: Clarify ExposedThing Property initialization

Commit:
Clarify ExposedThing default handlers vs script provided handlers. Editorial fixes.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
feat: align with latest scripting changes

* subscriptionId
* getThingDescription()
* optional  InteractionOptions

danielpeintner committed:
fix: readAllProperties() returns PropertyValueMap also

Commit:
Consistently use PropertyMap for Property interactions

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #187 from zolkis/master

Clarify ExposedThing default handlers vs script provided handlers. Editorial fixes.

Commit:
Add InteractionOptions parameters to handlers. Update Web IDL constructors.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge branch 'master' of github.com:w3c/wot-scripting-api

Commit:
Align params name between invokeAction and ActionHandler

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
feat: align with WebIDL changes

* add InteractionOptions

Commit:
Merge pull request #189 from zolkis/master

Update constructors. Add InteractionOptions to handlers.

danielpeintner committed:
Merge branch 'master' into next-api-2019

# Conflicts:
#	typescript/index.d.ts
#	typescript/package.json

danielpeintner committed:
add pr-preview

Commit:
Revert subscription ID change, see #183

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #194 from zolkis/master

Revert subscription ID change, see #183

danielpeintner committed:
fix: revert subscriptionID

danielpeintner committed:
Merge pull request #191 from danielpeintner/next-api-2019

Next api 2019

Commit:
Editorial: fix HTML errors

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge branch 'master' of github.com:w3c/wot-scripting-api

Commit:
Merge pull request #195 from zolkis/master

Fix HTML errors

Commit:
Release WD4

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge branch 'master' of github.com:w3c/wot-scripting-api

Commit:
Merge pull request #196 from zolkis/master

Release WD4

Commit:
tidy again with -i option

Commit:
fix This version and add Previous version

Commit:
fix doc type as Working Draft; tentative pubdate Oct 22

Commit:
fix CSS as W3C-WD

Commit:
published as a Working Draft

Commit:
fix this version url

Commit:
another Editor's Draft fixed as Working Draft

Commit:
add Editor IDs by w3cid

Commit:
add the link to public list of disclosures

Commit:
add Edito IDs by data-editor-id

Commit:
back-to-top uses #toc instad of #title

Commit:
fix broken link of the-start-method as file by adding # and make it #the-start-method

Commit:
fix broken link of the-next-method as file by adding # and make it #the-next-method

Commit:
use CR version URL for TD

Commit:
use CR version URL for Architecture

Commit:
updated pubdate Oct 28

Commit:
CR version URL for TD (again)

Commit:
not json-schema-4-validation but json-schema-for-validation

Commit:
publish only Overview.html

Commit:
feedback from wd4

Commit:
Merge pull request #197 from w3c/feedback-from-wd4

feedback from wd4

danielpeintner committed:
chore: label as final version for recent Scripting API publication

https://www.w3.org/TR/2019/WD-wot-scripting-api-20191028/

danielpeintner committed:
fix: typescript issue

see https://github.com/eclipse/thingweb.node-wot/issues/143

Commit:
Remove markdown from header

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #202: add formIndex to InteractionOptions

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #199: improve explanation for setting property write handlers

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #203 from zolkis/master

Fix markdown, add formIndex, explain write handlers

Commit:
use jquery-less ReSpec 

as it's lighter, see also
https://github.com/w3c/wot-thing-description/pull/872/files

Commit:
Fix #193: note about returning values from writes

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge branch 'master' of github.com:w3c/wot-scripting-api

Commit:
Merge pull request #205 from zolkis/master

Returning values from writes

Commit:
Add Web Platform cross reference

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Cross-link usage of errors and throw

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Editorial: update cross-referencing style

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #207 from zolkis/master

Editorial changes

Commit:
Add InteractionOptions to unobserveProperty and unsubscribeEvents

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Add InteractionData definition and usage

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Factor out validating Property value to a separate section

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Add property observe, unobserve and event handler support to ExposedThing; fixes #210 and #211

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Use 'any' in Property interactions but together with newly defined algorithms to parse and create interaction data

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix ConsumedThing example with using InteractionData

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
chore: remove snapshot

danielpeintner committed:
feat: add formIndex in InteractionOptions and InteractionData

see https://github.com/w3c/wot-scripting-api/pull/209

danielpeintner committed:
fix: mark all interaction options as optional

Commit:
Support streams in InteractionData and use InteractionData for returned data

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Improve explaining InteractionData

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Make value() a function. Update the read and write algorithms using streams.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
feat: align with latest PR updates

https://github.com/w3c/wot-scripting-api/pull/209

Commit:
Make value() fail early. Revise error messages.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Add an editor's note on Protocol Bindings

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #209 from zolkis/master

Add InteractionData

Commit:
Merge branch 'master' of github.com:w3c/wot-scripting-api into subscribehandlers

Commit:
Update ExposedThing algorithms

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Handle review comments

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Consolidate the use of InteractionData in ExposedThing. Improve the algorithms and add notes.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Clarify the use of InteractionData vs 'any'. Add pictures.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Align handler names

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Introduce the InteractionInput type and InteractionOutput interface

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Editorial: improve methods text on arguments

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Update figures to reflect function signatures better

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Use DataSchemaValue to differentiate from TD DataSchema

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix Web IDL and references for InteractionOptions fields

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #218 from zolkis/subscribehandlers

Add ExposedThing property observe, unobserve and event handler support

Commit:
Fix #215: make the API entry point a namespace. Apply Web IDL namespace rules.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Change namespace name back to WOT

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #220 from zolkis/namespace-wot

Fix #215: make the API entry point a namespace.

Commit:
Address #200: improve text on error handling

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #221 from zolkis/error-handling

Address #200: improve text on error handling

danielpeintner committed:
docs: minor typos/issues

# 7.13 The invokeAction() method

Point 6: "create interaction datasteps" --> "create interaction data steps"

# 8.6 Handling requests for reading multiple Poperties

Typo in header: "Poperties" --> "Properties"

Point 6, "Protocol Bindingsand" --> "Protocol Bindings and"

# 8.13 The setPropertyWriteHandler() method

"observeable" --> "observable"

# 8.20 The EventSubscriptionHandler callback

"with a options object" --> "with an options object "

# 9. The ThingDiscovery interface

"occured" --> "occurred"

relu91 committed:
Minor changes on chapters 4 and 5

Introduce @egekorkan review changes specified in the following
pdf document: https://github.com/w3c/wot-scripting-api/files/5017322/ScriptingAPI-review-egekorkan.pdf

Commit:
Merge pull request #229 from w3c/2020-08-03-minor-issues2

docs: minor typos/issues

relu91 committed:
Revert <dfn> elements and validation algorithm

relu91 committed:
Update validation algorithm with a default filling step

relu91 committed:
Update TD validation algorithm

Simplify so that we can build more complex steps from this solid ground

relu91 committed:
Add editor's note in TD validation

Commit:
Address #230: update first chapters
Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix ReSpec errors

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

relu91 committed:
Minor changes on chapters 4 and 5 (#231)

* Minor changes on chapters 4 and 5

Introduce @egekorkan review changes specified in the following
pdf document: https://github.com/w3c/wot-scripting-api/files/5017322/ScriptingAPI-review-egekorkan.pdf

* Revert <dfn> elements and validation algorithm

* Update validation algorithm with a default filling step

* Update TD validation algorithm

Simplify so that we can build more complex steps from this solid ground

* Add editor's note in TD validation

danielpeintner committed:
feat: update typescript definitions

danielpeintner committed:
minor: remove duplicated as

Commit:
Merge pull request #236 from w3c/danielpeintner-patch-1

minor: remove duplicated as

Commit:
Make refs to other WoT specs direct links

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Replace void with undefined as WebIDL spec changed

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

danielpeintner committed:
Merge pull request #235 from danielpeintner/2020-08-api-changes-ts

feat: update typescript definitions

Commit:
Merge pull request #234 from zolkis/master

Address #230: update first chapters

relu91 committed:
Merge remote-tracking branch 'upstream/master'

relu91 committed:
General text improvements and fixes

relu91 committed:
Rename create interaction data to create interaction request

relu91 committed:
Introduce writeAllProperties function

relu91 committed:
Fixes to read all/multiple algorithms

relu91 committed:
Update conform (un)observeProperty algo to (un)subscribe algo

Commit:
Add Cristiano Aguzzi as Editor

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

zolkis commented:
You can create a PR to an empty branch with only one file, the index.html, placed in a new directory named `review`.
Then you can make a PR against the Scripting API itself :)

egekorkan commented:
That does not work for some reason (why is this so not straighforward! :D) See here: https://github.com/w3c/wot-scripting-api/compare/master...egekorkan:empty

zolkis commented:
But if you create a new directory in Scripting and add a file there (named whatever, like ege-index.html), that whole file will be new addition in the PR, then everyone can comment on any line of it.

Commit:
Merge pull request #246 from zolkis/master

Add Cristiano Aguzzi as Editor

relu91 committed:
Revert "Introduce writeAllProperties function"

This reverts commit e5137dbc618b29f5856331173bdea66bbdefaa6a.
writeAllProperties method needs more discussion.

relu91 committed:
Add an editor's note for writeAllProperties

relu91 committed:
Fix minor review comments on #239

relu91 committed:
Fix create interaction request source type

danielpeintner committed:
example: fix/add missing bracket

Commit:
Merge pull request #249 from w3c/danielpeintner-patch-2

example: fix/add missing bracket

relu91 committed:
Update remove writeAllProperty section

add an editor's note at bottom of Full Web IDL

relu91 committed:
Fix follows-->follow

Commit:
Merge pull request #239 from relu91/review

My review feedback

danielpeintner committed:
chore: updates for Chapter 6 based on PDF comments from Ege (#250)

* chore: updates for Chapter 6 based on PDF comments from Ege

see https://github.com/w3c/wot-scripting-api/issues/230#issuecomment-668142199

* docs: change all appearances of "JSON schema" to "JSON Schema"

Commit:
Update README.md

Commit:
Remove Open Issue section, fixing #253. Also, autocorrect typos/whitespace errors.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Move ednote about writeMultipleProperties under the ConsumedThing WebIDL definition.

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

relu91 committed:
Update consume thing consume example

Simplify the example and closes #254

Commit:
Merge pull request #254 from zolkis/master

Remove Open Issue section, fixing #253.

Commit:
Merge pull request #255 from relu91/cosume_example

Update consume thing consume example

relu91 committed:
Update add more consume thing examples

relu91 committed:
Fix consumed thing examples

Commit:
Merge pull request #257 from relu91/cosume_example

Add more Consume Thing examples

Commit:
Fix #244: ConsumeThing constructor and consume() should do the same

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #243: make ReadHandler description more concise

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #251: improve the emitEvent() algorithm

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix #258: improve expose() algorithm

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Update README and rationale

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #260 from zolkis/readhandler-descr

Fix #243: make ReadHandler description more concise

Commit:
Merge pull request #261 from zolkis/improve-emitevent

Fix #251: improve the emitEvent() algorithm

Commit:
Add ednote about difference between consume() and constructing ConsumedThing

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Remove constraint about Forms in expose()

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #263 from zolkis/updates

Update README and rationale

Commit:
Merge pull request #262 from zolkis/expose-algo

Fix #258: improve expose() algorithm

Commit:
Merge pull request #259 from zolkis/consume-ctor

Fix #244: ConsumeThing constructor and consume() should do the same

Commit:
Fix #237 and #256: handle subscription errors

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Fix internal slots notation

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Factor out finding the unsubscribe Form

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Improve subscription error handling

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

relu91 committed:
Update: typescript definition support for no module systems.

Now ts definition file supports both module and no module system
environments.

Commit:
Merge pull request #264 from zolkis/error-handling

Fix #237 and #256: handle subscription errors

Commit:
Fix ReSpec header with new group field

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #266 from zolkis/master

Fix ReSpec header with new group field

danielpeintner committed:
Merge pull request #265 from relu91/type-def-proposal

Proposal for typescript definition file

danielpeintner committed:
chore: increase version to be able to publish it on NPM

relu91 committed:
Fix typescript definition module name

danielpeintner committed:
Merge pull request #267 from relu91/fix-ts-d

Fix typescript definition module name

danielpeintner committed:
feat: update version for NPM publish

danielpeintner committed:
feat: update TS definitions to "0.8.0-SNAPSHOT.5"

add "target": "es6"

Commit:
Improve error handling for ExposedThing:emitEvent()

Signed-off-by: Zoltan Kis <zoltan.kis@intel.com>

Commit:
Merge pull request #269 from zolkis/error-handling

Improve error handling for ExposedThing:emitEvent()

danielpeintner committed:
feat: remove section about "Stale TD Security Risk"

+ removes references for wot-security best practices and testing which are all part of wot-security document

danielpeintner committed:
Merge pull request #271 from danielpeintner/issue-252

feat: remove section about "Stale TD Security Risk"

relu91 committed:
Add editor's note about WoT interactions vs Interaction Affordances

relu91 committed:
Update affordances vs interactions

Improve description and remove editor's note

danielpeintner committed:
Merge pull request #272 from relu91/affordances

Add editor's note about WoT interactions vs Interaction Affordances
